File: universe/lib/models/widget.dart
enum WidgetType {
  music,
  poll,
  stopWatch,
  rate,
  location,
  code,
  question,
  answer,
}

class Widget {
  String title;
  String body;
  String data;
  WidgetType type;

  Widget({
    required this.title,
    required this.body,
    required this.data,
    required this.type,
  });
}

File: Universe_Backend/Data/Models/Widget.cs
using Microsoft.EntityFrameworkCore;

namespace Universe_Backend.Data.Models;

public enum WidgetType
{
    Music,
    Poll,
    StopWatch,
    Rate,
    Location,
    Code,
    Question,
    Answer,
}

[Owned]
public class Widget
{
    public required WidgetType Type { get; set; }
    public required string Title { get; set; }
    public required string Body { get; set; }
    public required string Data { get; set; }
}
